src/hb-ot-cff2-table.hh

```diff
@@ -0,0 +1,386 @@
+  hb_vector_t <float> familyOtherBlues;
+  hb_vector_t <float> stemSnapH;
+  hb_vector_t <float> stemSnapV;
+};
+
+struct CFF2PrivateDictOpSet
+{
+  static inline bool process_op (const ByteStr& str, unsigned int& offset, unsigned char byte, Stack& stack, CFF2PrivateDictValues& val)
+  {
+    switch (byte) {
+      case OpCode_BlueValues:
+        if (unlikely (!stack.check_pop_delta (val.blueValues, true)))
+          return false;
+        break;
+      case OpCode_OtherBlues:
+        if (unlikely (!stack.check_pop_delta (val.otherBlues, true)))
+          return false;
+        break;
+      case OpCode_FamilyBlues:
+        if (unlikely (!stack.check_pop_delta (val.familyBlues, true)))
+          return false;
+        break;
+      case OpCode_FamilyOtherBlues:
+        if (unlikely (!stack.check_pop_delta (val.familyOtherBlues, true)))
+          return false;
+        break;
+      case OpCode_StdHW:
+        if (unlikely (!stack.check_pop_real (val.stdHW)))
+          return false;
+        break;
+      case OpCode_StdVW:
+        if (unlikely (!stack.check_pop_real (val.stdVW)))
+          return false;
+        break;
+      case OpCode_BlueScale:
+        if (unlikely (!stack.check_pop_real (val.blueScale)))
+          return false;
+        break;
+      case OpCode_BlueShift:
+        if (unlikely (!stack.check_pop_real (val.blueShift)))
+          return false;
+        break;
+      case OpCode_BlueFuzz:
+        if (unlikely (!stack.check_pop_real (val.blueFuzz)))
+          return false;
+        break;
+      case OpCode_StemSnapH:
+        if (unlikely (!stack.check_pop_delta (val.stemSnapH)))
+          return false;
+        break;
+      case OpCode_StemSnapV:
+        if (unlikely (!stack.check_pop_delta (val.stemSnapV)))
+          return false;
+        break;
+      case OpCode_LanguageGroup:
+        if (unlikely (!stack.check_pop_int (val.languageGroup)))
+          return false;
+        break;
+      case OpCode_ExpansionFactor:
+        if (unlikely (!stack.check_pop_real (val.expansionFactor)))
+          return false;
+        break;
+      case OpCode_Subrs:
+        if (unlikely (!check_pop_offset (stack, val.subrsOffset)))
+          return false;
+        break;
+      case OpCode_blend:
+        // XXX: TODO
+        break;
+
+      default:
+        return false;
+    }
+    return true;
+  }
+};
+
+typedef Interpreter<CFF2TopDictOpSet, CFF2TopDictValues> CFF2TopDict_Interpreter;
+typedef Interpreter<CFF2FontDictOpSet, CFF2FontDictValues> CFF2FontDict_Interpreter;
+typedef Interpreter<CFF2PrivateDictOpSet, CFF2PrivateDictValues> CFF2PrivateDict_Interpreter;
+
+}; /* namespace CFF */
+
+namespace OT {
+
+using namespace CFF;
+
+struct cff2
+{
+  static const hb_tag_t tableTag        = HB_OT_TAG_cff2;
+
+  inline bool sanitize (hb_sanitize_context_t *c) const
+  {
+    TRACE_SANITIZE (this);
+    return_trace (c->check_struct (this) &&
+                  version.sanitize (c) &&
```

Comment:
No need to call sanitize on version since it's of static size. The check_struct above will cover it.

Reply:
fixed